bin/node/runtime/src/impls.rs

```diff
@@ -111,6 +111,29 @@ impl ProposalProvider<AccountId, Hash, RuntimeCall> for AllianceProposalProvider
 	}
 }
 
+pub struct Hasher;
+impl pallet_pov_limit::Hasher for Hasher {
+	type Out = ();
+
+	fn hash(_: &[u8]) -> Self::Out {
+		// this calculates factorial. should be replaced by a hashing algorithm
+		// but `Blake2Hasher` cannot be used here since this is no_std.
+		let num: u64 = 20;
+		match num {
+			0 => 1,
+			1.. => (1..num + 1).product(),
+		};
+	}
+}
+
+pub struct AccountReader;
+impl pallet_pov_limit::Reader for AccountReader {
+	fn read<T: pallet_pov_limit::Config>(_: &[u8]) -> Weight {
+		// should actually read something.
```

Comment:
@ggwpez I am not totally sure what you meant by reading accounts and how we would do that.

Reply:
Yes maybe that was not such a good idea after all. My thought was to have an easy way to read large number of data.  
BUT not all chains even have that many accounts. We probably have to use a normal `StorageMap` and initialize it once and then use that to read from.  
This would be more predictable and controlled. Something like `Map<u32, [u8; 32]>` maybe. We could also opt for a double-map with different bucket sizes since multiple reads will probably have more PoV overhead than a single larger one.  

You can probably finish the pallet first for the compute part and then do the PoV part analogous. I will try to find the time in the next few days to have a look.